allennlp/nn/module.py

```diff
@@ -0,0 +1,50 @@
+from typing import List, Optional
+
+from overrides import overrides
+import torch
+
+from allennlp.nn.util import (
+    _check_incompatible_keys,
+    _IncompatibleKeys,
+    StateDictType,
+    load_state_dict_distributed,
+)
+
+
+class Module(torch.nn.Module):
+    """
+    This is just `torch.nn.Module` with some extra functionality.
+    """
+
+    def _post_load_state_dict(self, missing_keys: List[str], unexpected_keys: List[str]) -> None:
+        """
+        Subclasses can override this and potentially modify `missing_keys` or `unexpected_keys` in place.
+        """
+        pass
```

Comment:
> Can we have them return two lists instead? Let's keep functions pure if it's not too much of a hassle.

👍

> Also, should those be sets, not lists?

Yes, but PyTorch uses lists instead of sets here, so I think it's better to be consistent.